skl2onnx/operator_converters/AdaBoost.py

```diff
@@ -0,0 +1,208 @@
+    sub_result_name = scope.get_unique_variable_name('sub_result')
+    samme_proba_name = scope.get_unique_variable_name('samme_proba')
+
+    container.add_initializer(
+        inverted_n_classes_name, onnx_proto.TensorProto.FLOAT,
+        [], [1. / n_classes])
+    container.add_initializer(
+        n_classes_minus_one_name, onnx_proto.TensorProto.FLOAT,
+        [], [n_classes - 1])
+
+    container.add_node(
+        'Clip', proba_name, clipped_proba_name,
+        name=scope.get_unique_operator_name('Clip'),
+        min=np.finfo(float).eps)  # max=None)
+    container.add_node(
+        'Log', clipped_proba_name, log_proba_name,
+        name=scope.get_unique_operator_name('Log'))
+    container.add_node(
+        'ReduceSum', log_proba_name, reduced_proba_name, axes=[1],
+        name=scope.get_unique_operator_name('ReduceSum'))
+    apply_reshape(scope, reduced_proba_name,
+                  reshaped_result_name, container,
+                  desired_shape=(-1, 1))
+    apply_mul(scope, [reshaped_result_name, inverted_n_classes_name],
+              prod_result_name, container, broadcast=1)
+    apply_sub(scope, [log_proba_name, prod_result_name],
+              sub_result_name, container, broadcast=1)
+    apply_mul(scope, [sub_result_name, n_classes_minus_one_name],
+              samme_proba_name, container, broadcast=1)
+    return samme_proba_name
+
+
+def _normalise_probability(scope, container, operator, proba_names_list,
+                           model):
+    est_weights_sum_name = scope.get_unique_variable_name('est_weights_sum')
+    summation_prob_name = scope.get_unique_variable_name('summation_prob')
+    div_result_name = scope.get_unique_variable_name('div_result')
+    exp_operand_name = scope.get_unique_variable_name('exp_operand')
+    exp_result_name = scope.get_unique_variable_name('exp_result')
+    reduced_exp_result_name = scope.get_unique_variable_name(
+        'reduced_exp_result')
+    normaliser_name = scope.get_unique_variable_name('normaliser')
+    zero_scaler_name = scope.get_unique_variable_name('zero_scaler')
+    comparison_result_name = scope.get_unique_variable_name(
+        'comparison_result')
+    cast_output_name = scope.get_unique_variable_name('cast_output')
+    zero_filtered_normaliser_name = scope.get_unique_variable_name(
+        'zero_filtered_normaliser')
+    mul_operand_name = scope.get_unique_variable_name('mul_operand')
+    cast_normaliser_name = scope.get_unique_variable_name('cast_normaliser')
+
+    container.add_initializer(
+        est_weights_sum_name, onnx_proto.TensorProto.FLOAT,
+        [], [model.estimator_weights_.sum()])
+    container.add_initializer(
+        mul_operand_name, onnx_proto.TensorProto.FLOAT,
+        [], [1. / (model.n_classes_ - 1)])
+    container.add_initializer(zero_scaler_name,
+                              onnx_proto.TensorProto.INT32, [], [0])
+
+    container.add_node('Sum', proba_names_list,
```

Comment:
for some common op like 'Sum', should we create apply_sum function for it?

Reply:
My understanding was operators Added with apply_* have different opset versions.

Reply:
@xadupre @prabhat00155 , we never know which op will be updated in the future and don't want to update the existing code for a new versioned op on adding a new apply_xxx. So understanding is the frequency of op.